File: tjhsst comp sci 2/pyKarel Student Code/Lab10.py
#Name: Vivian and Karthik
#Date:9/16/2019

from pyKarel import *
from Athlete import Athlete

maze = raw_input("What maze will you use?(maze1,maze2,maze3) ")

wld = World(maze,width=10,height=10)

karel = Athlete(wld,1,1)

active = True
while active:
   if not karel.frontIsClear():
      clear = 0
      if karel.leftIsClear():
         clear += 1
         
      if karel.rightIsClear():
         clear += 1
         
      if clear == 2:
         karel.turnRight()
         
      elif clear == 1:
         if karel.leftIsClear():
            karel.turnLeft()
         
         if karel.rightIsClear():
            karel.turnRight()
      
      else:
         karel.turnAround()
         
   else:     
      if karel.rightIsClear():
         karel.turnRight()
   
   if not karel.nextToABeeper():        
      karel.move()
   
wld.mainloop()
      

File: tjhsst comp sci 2/Java/Units/Unit1/Lab10.java
// Name: Vivian Feng             March 17,2020

import edu.fcps.karel2.Robot;
import edu.fcps.karel2.Display;
import javax.swing.JOptionPane;

public class Lab10
{
   public static void followWallsRight(Athlete arg)
   {
      while (arg.nextToABeeper() == false)
      {  //decide what to do if front has wall
         if (arg.frontIsClear() == false){
            int clear = 0;
            //checks left and right for walls
            if (arg.leftIsClear()){
               clear ++;
            }
            if ( arg.rightIsClear())
            {
               clear ++;
            }
            //turns right is possible. otherwise, turn left
            if (clear==2)
               arg.turnRight();
            
            else if (clear==1)
            {
               if (arg.rightIsClear())
                  arg.turnRight();
                  
               else
                  arg.turnLeft();
                  
            }else{
               arg.turnAround();
            }
         }else{
            if (arg.rightIsClear())
               arg.turnRight();
               
         }
         arg.move();
      }
   }
   
   public static void main(String[] args){
      String filename = JOptionPane.showInputDialog("Which maze(maze1,maze2,maze3)? ");
      Display.openWorld("maps/"+filename+".map");
      Display.setSize(10,10);
      Display.setSpeed(5);
      Athlete karel = new Athlete();
      followWallsRight(karel);
   }
}